File: python_backend/app/application/use_cases/list_personas.py
"""List personas for a user."""
from __future__ import annotations

from app.application.ports.personas_repository import PersonasRepository
from app.domain.entities.persona import Persona


class ListPersonasUseCase:
    def __init__(self, repo: PersonasRepository) -> None:
        self.repo = repo

    def execute(self, *, user_id: str) -> list[Persona]:
        return self.repo.list_for_user(user_id)

File: apps/web/src/application/use-cases/list-personas.ts
/**
 * ListPersonasUseCase — fetch the user's personas catalog.
 */

import type { PersonasApiPort } from "@/application/ports/personas-api.port";
import type { Persona } from "@/domain/entities/persona";

export class ListPersonasUseCase {
  constructor(private readonly api: PersonasApiPort) {}

  execute(): Promise<Persona[]> {
    return this.api.list();
  }
}
